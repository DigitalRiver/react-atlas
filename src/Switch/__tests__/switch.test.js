import React from "react";
import { mount } from "enzyme";
import { Switch } from "../index";

import renderer from "react-test-renderer";

describe("Test switch component render", () => {
  it("Render correctly", () => {
    const tree = renderer
      .create(
        <Switch
          className={"Class"}
          disabled={false}
          hidden={false}
          checked={false}
          name={"Name"}
          id={"ID"}
          onColor={"black"}
          onClick={() => {}}
          offColor={"white"}
          buttonColor={"grey"}
          small={false}
          medium
          large={false}
          onChange={() => {}}
          onBeforeChange={() => {}}
          inline={false}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Testing switch component", () => {
  it("Change Props", () => {
    const swich = mount(
      <Switch
        className={"Class"}
        disabled={false}
        hidden={false}
        checked={false}
        name={"Name"}
        id={"ID"}
        onColor={"black"}
        onClick={() => {}}
        offColor={"white"}
        buttonColor={"grey"}
        small={false}
        medium
        large={false}
        onChange={() => {}}
        onBeforeChange={() => {}}
        inline={false}
      />
    );

    swich.setProps({ "name": "NewName" });

    swich.setProps({ "checked": true });
    expect(swich.state().checked).toEqual(true);

    swich.setProps({ "disabled": true });
    expect(swich.state().disabled).toEqual(true);
  });
});
