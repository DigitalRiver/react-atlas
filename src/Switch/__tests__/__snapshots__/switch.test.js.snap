// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test switch component render Render correctly 1`] = `
<div
  className="Class switch"
>
  <div
    className="sliderMedium sliderEnabledOffColor"
    onClick={[Function]}
    style={undefined}
  >
    <input
      buttonColor="grey"
      checked={false}
      className="inputMedium"
      disabled={false}
      hidden={false}
      id="ID"
      name="Name"
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <div
      className="handleMedium handleEnabledColor"
      style={
        Object {
          "background": "grey",
        }
      }
    />
    <div
      className="onColorMedium"
      style={
        Object {
          "background": "black",
        }
      }
    />
  </div>
</div>
`;
